src/FwdState.cc

```diff
@@ -871,25 +909,35 @@ FwdState::secureConnectionToPeerIfNeeded()
     // as is
     const bool needTlsToOrigin = !p && request->url.getScheme() == AnyP::PROTO_HTTPS && !clientFirstBump;
 
-    if (needTlsToPeer || needTlsToOrigin || needsBump) {
+    if (needTlsToPeer || needTlsToOrigin || needsBump)
+        return advanceDestination("secure connection to peer", conn, [this,&conn] {
+            secureConnectionToPeer(conn);
+        });
+
+    // if not encrypting just run the post-connect actions
+    successfullyConnectedToPeer(conn);
+}
+
+/// encrypts an established TCP connection to peer (including origin servers)
+void
+FwdState::secureConnectionToPeer(const Comm::ConnectionPointer &conn)
+{
+    { // TODO: Remove this diff reduction before commit
```

Comment:
We should not forget to remove this, but let's not remove it yet.

Reply:
commit 62df15ad